store/user.js

```diff
@@ -9,11 +15,35 @@ class User {
 
   email = null
 
-  retrieveUser = async () => {}
+  url = null
+
+  organisationId = null
+
+  affiliationUrl = null
+
+  permissionsUrl = null
 
   constructor(rootStore) {
     this.rootStore = rootStore
   }
+
+  @action
+  logIn = async () => {
+    await this.retrieveUser()
+  }
+
+  @action
+  retrieveUser = async () => {
+    const user = await apiRequest('/user', 'GET', {}, {}, true)
+    this.rootStore.changeDataProvider(user.organisationId)
+    Object.assign(this, user)
+  }
+
+  @action
+  loadUserPermissions = async () => {
+    // if (!this.permissionsUrl) return
+    // const permission = await apiRequest(this.permissionsUrl)
+  }
```

Comment:
I am not sure we need explicit user permissions for this moment but whatever you wish. If you want to go that way, try to localise permission tags as much as possible because it might be extremely hard to rename something at some point.

Reply:
We need it for repository select, don't we?

Reply:
I assumed that `/users/{id}/data-providers` returns the list of only data providers that the user can edit. So, in this case, we can implicitly check existence of an ID in this list.

Reply:
I did not know about it. My fault. I will make changes accordingly.

Reply:
The endpoint does not work correctly. You will need to add a workaround before @valerapro makes changes. Here is what I got till this moment:

```js
import apiRequest from 'api'

const fetchSuggestions = async (userId, query) => {
  const { data } = await apiRequest(
    `/users/${userId}/data-providers`,
    'GET',
    {
      q: query,
    },
    {},
    true
  )

  return data.organisation.repositories
}
```